org-ql-completing-read.el

```diff
@@ -137,10 +137,15 @@ single predicate)."
                  (marker) (if-let (it (org-entry-get marker "TODO"))
                               (concat (propertize it 'face (org-get-todo-face it)) " ")
                             ""))
+                (commit-marker (candidate)
+                  (let ((marker (gethash candidate table)))
+                    (add-text-properties 0 (length candidate)
+                     (list 'marker marker)
```

Comment:
Since this is, perhaps unintentionally, a public-facing name, I'd suggest giving the text property the `org-ql` prefix. And you could call it `org-ql--marker`, with a doubly hyphen to discourage people from using it.

Reply:
A good point--it should have a package-name prefix--but why should people be discouraged from using it?

Reply:
Well, maybe "discourage people from using it" is a little strong. I really just meant "discourage people from complaining about any future changes to its name or associated value". 😄

Reply:
Well, upon further thinking about this, I wonder if we should go the opposite direction.  Consider Org Agenda buffers (and org-ql-view buffers, which imitate them), in which each line has text properties with markers pointing to the associated heading.  This is essentially an established API.  Why not embrace it here as well?  Why not use the same property name (since it's the same thing: a heading marker)?  Then packages like Embark could be compatible with both these org-ql-completing-read results and Org Agenda buffers and org-ql-view buffers.

Reply:
I didn't know org agenda buffers had those markers in text properties! In that case, you should definitely just adopt that as an API.

Reply:
I've changed the property to be `org-marker`, which is also what the headlines in the org agenda use.